### PR TITLE
Add failed status for assignments and retool machine execution logic

### PR DESCRIPTION
**Pull request by jrbotros**
Prevents multiple machines from picking up the same task and handles execution retries correctly.

### CODE REVIEW

File: orchestra/machine_tasks.py [hunk 1]
```diff
@@ -26,31 +32,46 @@ def execute(project_id, step_slug):
...
+        task_assignment.save()
+        return
     task_assignment.status = TaskAssignment.Status.SUBMITTED
     task_assignment.in_progress_task_data = task_data
     task_assignment.save()
-    task.status = Task.Status.COMPLETE
-    task.save()
-
-    create_subsequent_tasks(project)
+    if task.project.status == Project.Status.ABORTED:
```
Comment (marcua) [inline orchestra/machine_tasks.py:76]: `# If a long-running task's project was aborted while running, we ensure the aborted state on the task.`

File: orchestra/machine_tasks.py [hunk 2]
```diff
@@ -1,5 +1,9 @@
 from importlib import import_module
```
Comment (marcua) [inline orchestra/machine_tasks.py:1]: chatted offline about some tests

Comment (jrbotros) [inline orchestra/machine_tasks.py]: Test forthcoming, but will land this first!

File: orchestra/machine_tasks.py (hunk 1 above)
Comment (jrbotros) [inline orchestra/machine_tasks.py:76]: :white_check_mark:

**Comment by marcua** [comment]
the code lgtm---some unit tests around `execute` that mock out a bunch of the external services would be helpful

**Comment by marcua** [comment]
lgtm!

File: orchestra/machine_tasks.py (hunk 2 above)
Comment (jrbotros) [inline orchestra/machine_tasks.py:1]: :white_check_mark:

**Pull request merged**